### DISCUSSION TITLE
SVG support in `fastn`

### DISCUSSION THREAD

**Discussion by gsalunke** [fastn-stack/fastn#311] [Ideas & RFCs]
We want to support SVG in FTD.

We can have two modes:

```ftd
-- ftd.svg foo:

<body of svg>
```

Now onwards people can use `foo` as `src` for a `ftd.image`:

```
-- ftd.image:
src: $foo
```

We should also support all the SVG elements as FTD kernel elements.

**Comment by amitu** [comment]
https://ahelwer.ca/post/2023-04-06-dark-mode/

**Comment by amitu** [comment]
@harshdoesdev we should support this as well.

We will allow:

```ftd
-- ftd.svg:
src.external: <optional src, if specified, content would be inlined>

<option svg body>
```

Either `src` or `body` should be passed to `ftd.svg`. `ftd.svg` is a new kernel element. 

We will also support `ftd.inline-svg` type:

```
-- ftd.inline-svg login:
src: <optional>

<optional inline svg>

-- ftd.svg:
svg.inline: $login
```
